### PR TITLE
Added bootstrap 4 theme

### PR DESCRIPTION
**Pull request by ryanbeymer**
#1486 - First pass at adding bootstrap 4 support

### CODE REVIEW

**Comment by ufo911** [comment]
How long will it take to get BS4 fully working ?

**Comment by ryanbeymer** [comment]
PR is ready and passed build.

**Comment by ufo911** [comment]
@ryanbeymer can I test it  ?  how to..

**Comment by ryanbeymer** [comment]
https://github.com/ryanbeymer/flask-admin/tree/master/flask_admin/templates

**Comment by ufo911** [comment]
@ryanbeymer you may also add static bootstrap 4 folder

**Comment by ryanbeymer** [comment]
@ufo911 thx for catching that, I had it locally, forgot to copy that over.

**Comment by ufo911** [comment]
other_errors
more errors

**Comment by ryanbeymer** [comment]
What's one of those paths?

**Comment by ufo911** [comment]
capture

**Comment by ufo911** [comment]
can you also check this ? https://github.com/flask-admin/flask-admin/issues/1676

**Comment by ryanbeymer** [comment]
Looks like you don't have the files, have you updated the package? 

Can you check to make sure they are there?

**Comment by ufo911** [comment]
@ryanbeymer got it worked but other problem
capture

**Comment by ryanbeymer** [comment]
Is that whole `.hide` block supposed to be hidden?

**Comment by ufo911** [comment]
It have to be hidden until pressing add categories

**Comment by ufo911** [comment]
pressing add categories also does not work

**Comment by petrus-jvrensburg** [comment]
@ryanbeymer @ufo911 Let me review this. Is there anything that you know of that still needs work?

**Comment by petrus-jvrensburg** [comment]
I merged this into a new `bootstrap4` branch, where I'm seeing some issues when trying out the new templates on example apps.

- SQLAlchemy2 example: the 'With selected' button doesn't line up with the 'List' and 'Create' tabs
- Simple views example: The dropdown for "Test" views isn't styled correctly
- Auth example: Once logged-in, the 'logout' button in the dropdown on the right is not showing

Any help in fixing these, or other bugs would be appreciated. 

Also, if anyone can add Bootswatch support for the bootstrap4 templates, in the same way as it has been done for the other bootstraps, then that would be great.

screen shot 2018-09-26 at 13 35 20
screen shot 2018-09-26 at 13 32 44
screen shot 2018-09-26 at 13 30 11

**Comment by waylonwang** [comment]
I tested under the bootstrap4 branch and found the modal dialog not showing when create_modal = True

**Comment by waylonwang** [comment]
To display modal forms in the bootstrap4 theme you need to make the following adjustments:
1. Add bs4_modal.js under /static/admin/js
2. Include the following code in bs4_modal.js:
```
$('body').on('click.modal.data-api', '[data-toggle="modal"]', function () {
     $($(this).data("target") + ' .modal-content').load($(this).attr('href'));
});
```
3. Add a reference to /templates/bootstrap4/admin/model/list.html:
`<script src="{{ admin_static.url(filename='admin/js/bs4_modal.js', v='1.0.0') }}"></script>`
4. Change jquery-3.3.1.slim.min.js in /templates/bootstrap4/admin/base.html to jquery.min.js

**Comment by QuietCoder1007** [comment]
Does Flask-Admin work with Bootstrap 4 now?

File: flask_admin/static/vendor/fontawesome-free-5.1.0-web/less/_animated.less [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Animated Icons
+// --------------------------
+
+.@{fa-css-prefix}-spin {
+  animation: fa-spin 2s infinite linear;
+}
+
+.@{fa-css-prefix}-pulse {
+  animation: fa-spin 1s infinite steps(8);
+}
```
Comment (petrus-jvrensburg) [inline flask_admin/static/vendor/fontawesome-free-5.1.0-web/less/_animated.less:10]: don't do this

**Comment by daniel-butler** [comment]
Anything I can do to help push this along?

**Comment by michaelbukachi** [comment]
Hello. I also want to help out on this. What's pending?

**Pull request closed without merging**